### PR TITLE
Use --config-json for configuration

### PR DESCRIPTION
**Pull request by wpoely86**
@boegel have a look

### CODE REVIEW

File: run_lmod_cache.py [hunk 1]
```diff
@@ -34,23 +35,34 @@ def run_cache_create(modules_root):
...
-        'cache_dir': '/apps/gent/lmodcache/',
-        'cache_timestamp': '/apps/gent/lmodcache/timestamp',
-    }
+    lmod_cmd = os.environ.get("LMOD_CMD", None)
+    if not lmod_cmd:
+        raise OSError("Cannot find $LMOD_CMD in the environment.")
+
+    ec, out = run_simple("%s bash --config-json" % lmod_cmd)
+    if ec != 0:
+        raise OSError("Failed to get Lmod configuration: %s", out)
```
Comment (boegel) [inline run_lmod_cache.py]: `RuntimeError`?

File: run_lmod_cache.py [hunk 2]
```diff
@@ -34,23 +35,34 @@ def run_cache_create(modules_root):
...
+        lmodconfig = json.loads(out)
+
+        config = {
+            'modules_root': lmodconfig['config']['mpath_root'],
+            'cache_dir': lmodconfig['cache'][0][0],
+            'cache_timestamp': lmodconfig['cache'][0][1],
+        }
+        logger.debug("Found Lmod config: %s", config)
+    except (ValueError, KeyError, IndexError, TypeError) as err:
+        raise OSError("Failed to parse the Lmod configuration: %s", err)
```
Comment (boegel) [inline run_lmod_cache.py]: also here `RuntimeError`?

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by boegel** [comment]
looks ok besides minor remark

I assume this is properly tested?

**Comment by wpoely86** [comment]
yes, tests pass and cache building still works

**Review by boegel — APPROVED** [review]

**Pull request merged**